### PR TITLE
8260571: Add PrintMetaspaceStatistics to print metaspace statistics upon VM exit

### PR DESCRIPTION
**Pull request by tstuefe**
I would like to backport this change to JDK-11 since this flag helps us analyzing metaspace problems, and beyond that its completely harmless.

The original patch does not properly apply, since the flag definition format in globals.hpp has changed since JDK11. Other than that, the patch is trivial.

Nightlies at SAP ran through with this patch for several weeks now.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260571](https://bugs.openjdk.java.net/browse/JDK-8260571): Add PrintMetaspaceStatistics to print metaspace statistics upon VM exit

### Reviewers
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/83/head:pull/83` \
`$ git checkout pull/83`

Update a local copy of the PR: \
`$ git checkout pull/83` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/83/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 83`

View PR using the GUI difftool: \
`$ git pr show -t 83`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/83.diff">https://git.openjdk.java.net/jdk11u-dev/pull/83.diff</a>

</details>

### CODE REVIEW

**Comment by RealCLanger** [comment]
Hi, @tstuefe. Any updates on this one?

**Comment by tstuefe** [comment]
> Hi, @tstuefe. Any updates on this one?

Its on my list..

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -1824,6 +1824,10 @@ define_pd_global(uint64_t,MaxRAM,                    1ULL*G);
           "class pointers are used")                                        \
           range(1*M, 3*G)                                                   \
                                                                             \
+  diagnostic(bool, PrintMetaspaceStatisticsAtExit, false,                   \
+                  "Print metaspace statistics upon VM exit.")               \
```
Comment (RealCLanger) [inline src/hotspot/share/runtime/globals.hpp:1828]: Could you fix the indentation of this line to match the formatting for the other flags around here?

File: src/hotspot/share/runtime/globals.hpp [hunk 2]
```diff
@@ -1824,6 +1824,10 @@ define_pd_global(uint64_t,MaxRAM,                    1ULL*G);
...
+                                                                            \
+                                                                            \
```
Comment (RealCLanger) [inline src/hotspot/share/runtime/globals.hpp:1830]: one empty line is enough, I'd say.

**Review by RealCLanger — CHANGES_REQUESTED** [review]
Overall looks good, I just have two formatting remarks which IMHO should be cleaned up.

Other than that, maybe you could enable Github Actions on your repository to get a test run after your next push.

**Comment by tstuefe** [comment]
> Overall looks good, I just have two formatting remarks which IMHO should be cleaned up.
> 
> Other than that, maybe you could enable Github Actions on your repository to get a test run after your next push.

Thanks Christoph. GHAs enabled, format changes done. Note that the patch cooked in our internal systems for about two months now, so I don't expect surprises.

**Comment by RealCLanger** [comment]
> > Overall looks good, I just have two formatting remarks which IMHO should be cleaned up.
> > Other than that, maybe you could enable Github Actions on your repository to get a test run after your next push.
> 
> Thanks Christoph. GHAs enabled, format changes done. Note that the patch cooked in our internal systems for about two months now, so I don't expect surprises.

Sure, LGTM now 👍

**Review by RealCLanger — APPROVED** [review]

**Comment by tstuefe** [comment]
> > > Overall looks good, I just have two formatting remarks which IMHO should be cleaned up.
> > > Other than that, maybe you could enable Github Actions on your repository to get a test run after your next push.
> > 
> > 
> > Thanks Christoph. GHAs enabled, format changes done. Note that the patch cooked in our internal systems for about two months now, so I don't expect surprises.
> 
> Sure, LGTM now

Thanks. I'll wait for the GHAs to finish, then push.

**Pull request closed without merging**